app/components/formatted_markdown_text.js

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Text} from 'react-native';
+import PropTypes from 'prop-types';
+import Renderer from 'commonmark-react-renderer';
+import {Parser} from 'commonmark';
+import {injectIntl, intlShape} from 'react-intl';
+import MarkdownLink from 'app/components/markdown/markdown_link';
+import {concatStyles, changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {getMarkdownTextStyles} from 'app/utils/markdown';
+
+const TARGET_BLANK_URL_PREFIX = '!';
+
+/*
+* Translations component with a similar API to <FormattedText> component except the message string
+* accepts markdown. It supports the following non-block-level markdown:
+* - *italic*
+* - **bold**
+* - `inline code`
+* - ~~strikethrough~~
+* - [link](http://example.com/)
+* - line\nbreaks
+*
+* Note: Line breaks (\n) in a defaultMessage parameter string must be surrounded by curly brackets {} in JSX. Example:
+* <FormattedMarkdownText id='my.example' defaultMessage={'first line\nsecond line'} theme={theme} />
+*/
+class FormattedMarkdownText extends React.PureComponent {
+    static propTypes = {
+        intl: intlShape.isRequired,
+        id: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired,
+        defaultMessage: PropTypes.string,
+        values: PropTypes.object,
+        style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
```

Comment:
This can use `CustomPropTypes.Style` from `app/utils/custom_prop_types` since I think styles from spreadsheets are stored as a number